pythran/pythonic/types/str.hpp

```diff
@@ -704,64 +704,63 @@ namespace std
     return pythonic::types::str(t[I]);
   }
 }
+#ifdef ENABLE_PYTHON_MODULE
+
+#include "pythonic/python/core.hpp"
 
 namespace pythonic
 {
-  python_to_pythran<types::str>::python_to_pythran()
-  {
-    static bool registered = false;
-    if (not registered) {
-      registered = true;
-      boost::python::converter::registry::insert(
-          &convertible, &construct, boost::python::type_id<types::str>());
-    }
-  }
 
-  void *python_to_pythran<types::str>::convertible(PyObject *obj_ptr)
-  {
-    if (!PyString_Check(obj_ptr))
-      return 0;
-    return obj_ptr;
-  }
+  template <>
+  struct to_python<types::str> {
+    static PyObject *convert(types::str const &v)
+    {
+      return PyString_FromStringAndSize(v.c_str(), v.size());
+    }
+  };
 
-  void python_to_pythran<types::str>::construct(
-      PyObject *obj_ptr,
-      boost::python::converter::rvalue_from_python_stage1_data *data)
-  {
-    void *storage = ((boost::python::converter::rvalue_from_python_storage<
-                         types::str> *)(data))->storage.bytes;
-    char *s = PyString_AS_STRING(obj_ptr);
-    new (storage) types::str(s, PyString_GET_SIZE(obj_ptr));
-    data->convertible = storage;
-  }
+  template <class S>
+  struct to_python<types::sliced_str<S>> {
+    static PyObject *convert(types::sliced_str<S> const &v)
+    {
+      return ::to_python(types::str(v));
+    }
+  };
+  template <>
+  struct to_python<char> {
+    static PyObject *convert(char l)
+    {
+      return PyString_FromStringAndSize(&l, 1);
+    }
+  };
 
-  PyObject *custom_pythran_string_to_str::convert(const types::str &v)
-  {
-    return PyString_FromStringAndSize(v.c_str(), v.size());
-  }
+  template <>
+  struct from_python<types::str> {
+    static bool is_convertible(PyObject *obj)
+    {
+      return PyString_Check(obj);
+    }
+    static types::str convert(PyObject *obj)
+    {
+      return {PyString_AS_STRING(obj), (size_t)PyString_GET_SIZE(obj)};
+    }
+  };
+}
+/* type inference stuff  {*/
+#include "pythonic/include/types/combined.hpp"
```

Comment:
This file is not in the include folder but type inference shoudl be there to compute correct function return types (Some use __combine in their types)